test/factory.ts

```diff
@@ -343,14 +385,36 @@ describe('factory', () => {
       let caught = false;
       let err: Error;
       try {
-        await factory.call(instance, 'foo' as 'run');
+        await factory.call(instance, 'foo' as GitHubReleaseMethod);
       } catch (e) {
         err = e;
         caught = true;
       }
       expect(caught).to.be.true;
       expect(err!.message).to.equal('No such method(foo) on GitHubRelease');
     });
+    it('calls a Manifest instance', async () => {
+      const instance = factory.manifest({
+        repoUrl: 'googleapis/simple-test-repo',
+      });
+      sandbox.stub(instance, 'pullRequest').resolves(32);
+      expect(await factory.call(instance, 'pullRequest')).to.eql(32);
+    });
+    it('errors with bad method on Manifest', async () => {
+      const instance = factory.manifest({
+        repoUrl: 'googleapis/simple-test-repo',
+      });
+      let caught = false;
+      let err: Error;
+      try {
```

Comment:
I'm to blame, because I think I started this pattern, but I've started moving towards using `assert.rejects` and `assert.throws` for this type of assertion, I think it's cleaner:

https://nodejs.org/api/assert.html#assert_assert_rejects_asyncfn_error_message

You just do:

```js
await assert.rejects(factory.call(instance, 'foo' as ManifestMethod));
```

Reply:
I've tried your syntax and a few other incantations per the docs but I always end up with a test failure and this output (as if the error is bubbling past the `await assert.rejects` somehow):

```
  1) factory
       call                                                                                                                                                                                                                                                                                                                            errors with bad method on Manifest:
     Error: No such method(foo) on Manifest                                                                                                                                                                                                                                                                                         at Object.call (build/src/factory.js:79:15)
          -> /usr/local/google/home/joeldodge/release-please/src/factory.ts:161:11
at Context.<anonymous> (build/test/factory.js:392:52)
          -> /usr/local/google/home/joeldodge/release-please/test/factory.ts:407:36
at processImmediate (internal/timers.js:439:21)
```

Reply:
i could give chai-as-promised [a shot...](https://stackoverflow.com/a/32790325/13802304)

Reply:
I went with a different pattern:

```typescript
expect(() => factory.call(instance, 'foo' as ManifestMethod)).to.throw(
  'No such method(foo) on Manifest'
);
```